pkg/discovery/dns/resolver_test.go

```diff
@@ -33,99 +33,102 @@ func (m mockHostnameResolver) LookupSRV(ctx context.Context, service, proto, nam
 			expectedErr:    nil,
 			resolver: &mockHostnameResolver{
 				resultIPs: map[string][]net.IPAddr{
-					dnsHostNoPort: {net.IPAddr{IP: net.ParseIP(ip)}},
+					"test.mycompany.com": {net.IPAddr{IP: net.ParseIP("192.168.0.1")}},
 				},
 			},
 		},
+		{
+			testName:       "error on dns lookup when no port is specified",
+			addr:           "test.mycompany.com",
+			qtype:          A,
+			expectedResult: nil,
+			expectedErr:    errors.Errorf("missing port in address given for dns lookup: test.mycompany.com"),
+			resolver:       &mockHostnameResolver{},
+		},
 		{
 			testName:       "multiple srv records from srv lookup",
-			addr:           srvHost,
-			qtype:          "dnssrv",
-			expectedResult: []string{ip + ":" + strconv.Itoa(port), ip2 + ":" + strconv.Itoa(port)},
+			addr:           "_test._tcp.mycompany.com",
+			qtype:          SRV,
+			expectedResult: []string{"192.168.0.1:8080", "192.168.0.2:8081"},
 			expectedErr:    nil,
 			resolver: &mockHostnameResolver{
 				resultSRVs: map[string][]*net.SRV{
-					srvHost: {
-						&net.SRV{Target: ip, Port: uint16(port)},
-						&net.SRV{Target: ip2, Port: uint16(port)},
+					"_test._tcp.mycompany.com": {
+						&net.SRV{Target: "192.168.0.1", Port: 8080},
+						&net.SRV{Target: "192.168.0.2", Port: 8081},
 					},
 				},
 			},
 		},
 		{
-			testName:       "error on dns lookup when no port is specified",
-			addr:           dnsHostNoPort,
-			qtype:          "dns",
+			testName:       "multiple srv records from srv lookup",
+			addr:           "_test._tcp.mycompany.com:8082",
+			qtype:          SRV,
+			expectedResult: []string{"192.168.0.1:8082", "192.168.0.2:8082"},
+			expectedErr:    nil,
+			resolver: &mockHostnameResolver{
+				resultSRVs: map[string][]*net.SRV{
+					"_test._tcp.mycompany.com": {
+						&net.SRV{Target: "192.168.0.1", Port: 8080},
+						&net.SRV{Target: "192.168.0.2", Port: 8081},
+					},
+				},
+			},
+		},
+		{
+			testName:       "error from SRV resolver",
+			addr:           "_test._tcp.mycompany.com",
+			qtype:          SRV,
 			expectedResult: nil,
-			expectedErr:    errors.Errorf("missing port in address given for dns lookup: %v", dnsHostNoPort),
-			resolver:       &mockHostnameResolver{},
+			expectedErr:    errors.Wrapf(errorFromResolver, "lookup SRV records \"_test._tcp.mycompany.com\""),
+			resolver:       &mockHostnameResolver{err: errorFromResolver},
 		},
 		{
 			testName:       "error on bad qtype",
-			addr:           dnsHostNoPort,
-			qtype:          invalidQtype,
+			addr:           "test.mycompany.com",
+			qtype:          "invalid",
 			expectedResult: nil,
-			expectedErr:    errors.Errorf("invalid lookup scheme %q", invalidQtype),
+			expectedErr:    errors.Errorf("invalid lookup scheme \"invalid\""),
 			resolver:       &mockHostnameResolver{},
 		},
 		{
-			testName:       "error from resolver",
-			addr:           srvHost,
-			qtype:          "dnssrv",
+			testName:       "error on bad qtype",
```

Comment:
This test looks like the one above? Maybe delete it, or use a valid `qtype` with an invalid address (maybe this was the intention of the test?)